### ISSUE TITLE
Regarding the issue of reproducing results

### ISSUE THREAD

**Issue by cainiaoxiaozhao** [Wu0409/DuPL#10]
hello！author:
    Thanks for your research. I want to know that my result in evaluation of CAM pseudo labels(use ImageNet-21k pretrained parameters) on voc'train has a difference in yours.yours is 76.0%,my 74.806%.And,my Semantic Segmentation Results in voc'val also has a difference in yours,yours 73.3%,my 72.090%，they differ by 1 percentage poin.
    this is my 
[train.log](https://github.com/user-attachments/files/17867518/train.log).Can you help me?

**Comment by Wu0409** [comment]
Hey, thank you for your interest in our project!

From your `train.log`, it seems that the pre-trained backbone you used was still ImageNet-1k instead of 21k: `backbone='deit_base_patch16_224`.  If you want to use 21k backbone, you should modify to `backbone='vit_base_patch16_224'`(probably you need to manually set the directory of the weight file). 

Hope that helps~

**Comment by cainiaoxiaozhao** [comment]
Very thanks for your reply.I have modified the pretrained parameters as  ImageNet-21k. I am not sure if such a modification will work.
![image](https://github.com/user-attachments/assets/5a028b98-464b-463d-8c81-f900b2e364c6)

**Comment by Wu0409** [comment]
This should work. I have reviewed the log file again, and I can confirm that you are using the 21k weights, as the performance of CAM_AUX is significantly better compared to the 1k weights.

Therefore, the performance gap is likely due to differences in the experimental environment. As mentioned in previous issues, such differences can lead to performance variations of 0.5-1.0%. I’m also confused about why the environment seems so sensitive, but due to time constraints, I haven’t investigated further so far. For now, in my opinion, the variation of 1% is tolerable, and you can proceed with your subsequent experiments if you want :)

PS: You can change the random settings to check whether there has large performance variation.

**Comment by cainiaoxiaozhao** [comment]
OK.Thank you very much. I used your uploaded training 21k weights to evaluate the pseudo labels on the VOC training set, and the mIoU and segmentation results  were also about one point lower. Next, I will try to retrain using PyTorch version 1.13.1 (I previously used PyTorch version 2.0.1), and then I will attempt to conduct experiments on the COCO dataset.

**Comment by Wu0409** [comment]
> OK.Thank you very much. I used your uploaded training 21k weights to evaluate the pseudo labels on the VOC training set, and the mIoU and segmentation results  were also about one point lower. Next, I will try to retrain using PyTorch version 1.13.1 (I previously used PyTorch version 2.0.1), and then I will attempt to conduct experiments on the COCO dataset.

You mean you tested the released weights of 21k, and obtain similar results? This seems weird because the inference stage is not involved any randomness. If so, I will check this issue in next few days.

**Comment by cainiaoxiaozhao** [comment]
yes,this is specific results as follow:
![image](https://github.com/user-attachments/assets/19d718ad-7df2-46c5-a71a-bdea303f15b9)

**Comment by Wu0409** [comment]
Update: we have checked the weight file, and the results are consistent with those reported in the paper:

+-----------------+-----------+-----------+--------+-------+
|      Class      | confusion | precision | recall |  iou  |
+=================+===========+===========+========+=======+
| bg              | 0.022     | 0.978     | 0.936  | 0.917 |
+-----------------+-----------+-----------+--------+-------+
| aeroplane       | 0.279     | 0.782     | 0.974  | 0.766 |
+-----------------+-----------+-----------+--------+-------+
| bicycle         | 0.730     | 0.578     | 0.903  | 0.544 |
+-----------------+-----------+-----------+--------+-------+
| bird            | 0.197     | 0.836     | 0.968  | 0.813 |
+-----------------+-----------+-----------+--------+-------+
| boat            | 0.542     | 0.648     | 0.947  | 0.626 |
+-----------------+-----------+-----------+--------+-------+
| bottle          | 0.117     | 0.895     | 0.920  | 0.830 |
+-----------------+-----------+-----------+--------+-------+
| bus             | 0.133     | 0.883     | 0.971  | 0.860 |
+-----------------+-----------+-----------+--------+-------+
| car             | 0.179     | 0.848     | 0.939  | 0.803 |
+-----------------+-----------+-----------+--------+-------+
| cat             | 0.035     | 0.966     | 0.947  | 0.917 |
+-----------------+-----------+-----------+--------+-------+
| chair           | 0.318     | 0.759     | 0.586  | 0.494 |
+-----------------+-----------+-----------+--------+-------+
| cow             | 0.062     | 0.942     | 0.935  | 0.884 |
+-----------------+-----------+-----------+--------+-------+
| table           | 0.159     | 0.863     | 0.826  | 0.730 |
+-----------------+-----------+-----------+--------+-------+
| dog             | 0.066     | 0.938     | 0.960  | 0.902 |
+-----------------+-----------+-----------+--------+-------+
| horse           | 0.083     | 0.923     | 0.920  | 0.855 |
+-----------------+-----------+-----------+--------+-------+
| motorbike       | 0.158     | 0.864     | 0.950  | 0.826 |
+-----------------+-----------+-----------+--------+-------+
| person          | 0.110     | 0.901     | 0.945  | 0.856 |
+-----------------+-----------+-----------+--------+-------+
| plant           | 0.698     | 0.589     | 0.930  | 0.564 |
+-----------------+-----------+-----------+--------+-------+
| sheep           | 0.034     | 0.967     | 0.958  | 0.928 |
+-----------------+-----------+-----------+--------+-------+
| sofa            | 0.402     | 0.713     | 0.845  | 0.630 |
+-----------------+-----------+-----------+--------+-------+
| train           | 0.518     | 0.659     | 0.952  | 0.638 |
+-----------------+-----------+-----------+--------+-------+
| tvmonitor       | 0.786     | 0.560     | 0.949  | 0.543 |
+-----------------+-----------+-----------+--------+-------+
| average_metrics | 0.268     | 0.814     | 0.917  | **0.758** |
+-----------------+-----------+-----------+--------+-------+

The complete log file can be viewed [here](https://github.com/user-attachments/files/17896394/results.log).

**Comment by Wu0409** [comment]
No further updates.
